src/contour/ContourGuiApp.cpp

```diff
@@ -269,7 +269,27 @@ bool ContourGuiApp::loadConfig(string const& target)
         }
         else
         {
-            shell.program = flags.verbatim.front();
+            auto frontCommand = flags.verbatim.front();
+
+            // check if this is a file
+            if (fs::exists(frontCommand) && fs::is_regular_file(frontCommand))
+            {
+                // check if this is an executable file
+                if ((fs::status(frontCommand).permissions() & fs::perms::owner_exec) != fs::perms::none)
+                    shell.program = frontCommand;
+                else // find a path to file and open shell in this path
+                    shell.workingDirectory = fs::path(frontCommand).parent_path();
+            }
+            else if (fs::exists(frontCommand) && fs::is_directory(frontCommand))
+            {
+                shell.workingDirectory = fs::path(frontCommand);
+            }
+            else
+            {
+                errorLog()("Do not know what to do with `{}` will use it as a program", frontCommand);
```

Comment:
```
   p - execlp(), execvp(), execvpe()
       These  functions  duplicate  the  actions of the shell in searching for an executable file if the specified filename does not
       contain a slash (/) character.  The file is sought in the colon-separated list of directory pathnames specified in  the  PATH
       environment  variable.   If  this  variable isn't defined, the path list defaults to a list that includes the directories re‐
       turned by confstr(_CS_PATH) (which typically returns the value "/bin:/usr/bin") and possibly also the current working  direc‐
       tory; see NOTES for further details.
```

Looks like we should allow non-fully-qualified paths. Because we use `execvp` and that's totally fine with, say, executing `ls`, resolving to `/bin/ls` iff found in $PATH.

I think what would work best is:

1. check if it's a (relative or absolute) directory path, then take this as working directory and default shell

2.) alternatively assume it's a program, give execvp (through Process) a try as usual. we'll fall-back to configured login shell otherwise.